### DISCUSSION TITLE
ToolTip and Legend GeometryStrokes and GeometryFill not updating correctly

### DISCUSSION THREAD

**Discussion by rocinn** [Live-Charts/LiveCharts2#1632] [Q&A] [answered]
I am currently using LiveChartsCore.SkiaSharpView.Avalonia version 2.0.0-rc3

![image](https://github.com/user-attachments/assets/e756b4d9-fba1-4b93-81cd-37a84ea168ff)
![Clip_2024-10-09_12-39-33](https://github.com/user-attachments/assets/196b59e2-44b2-4e0c-a006-55d6c283f0ca)

**Comment by beto-rodriguez** [comment]
Hi

See https://github.com/beto-rodriguez/LiveCharts2/discussions/1507 for the answer, this seems a constant confusion, I think in a future version the legend should use the Stroke, if the GeometryStroke is null.

**Reply by rocinn** [reply]
> Hi
> 
> See #1507 for the answer, this seems a constant confusion, I think in a future version the legend should use the Stroke, if the GeometryStroke is null.

![Clip_2024-10-09_12-20-47](https://github.com/user-attachments/assets/dc620508-10e8-4492-872d-03a331923ed3)
![PixPin_2024-10-09_12-25-06](https://github.com/user-attachments/assets/3ad3874a-c864-41b6-8b29-9ecb35a626d6)

**Reply by beto-rodriguez** [reply]
The property used in the tooltip/legend is the `GeometryStroke`, in your case is null, that is why it is using the default.

**Reply by beto-rodriguez** [reply]
Try, `GeometrySize = 0` also to make the geometry invisible (instead of your null approach)

**Reply by rocinn** [reply]
> The property used in the tooltip/legend is the `GeometryStroke`, in your case is null, that is why it is using the default.

![Clip_2024-10-09_12-30-35](https://github.com/user-attachments/assets/008fe09c-8596-48fa-94be-a1d6c079f1e8)
![PixPin_2024-10-09_12-30-06](https://github.com/user-attachments/assets/3a22d377-5363-4e7a-81b5-03bf995bf57e)

**Reply by rocinn** [reply]
> Try, `GeometrySize = 0` also to make the geometry invisible (instead of your null approach)

![Clip_2024-10-09_12-34-06](https://github.com/user-attachments/assets/b045c9be-1ab5-4f18-94e7-ac1d25d04cb0)
![PixPin_2024-10-09_12-33-42](https://github.com/user-attachments/assets/332d8062-49f3-456c-8a08-c4935274a4c0)

**Reply by rocinn** [reply]
> Hi
> 
> See #1507 for the answer, this seems a constant confusion, I think in a future version the legend should use the Stroke, if the GeometryStroke is null.

I have checked issue #1507, but it does not resolve the problem; it seems the control caches the color based on the name.

**Reply by beto-rodriguez** [reply]
mm, looks strange.

When you click the check box, are you toggling the `IsVisible` property?

I will try to reproduce, but it would be easier for me if you could share some of your code, it seems that this is a different issue.

**Reply by rocinn** [reply]
> mm, looks strange.
> 
> When you click the check box, are you toggling the property?`IsVisible`
> 
> I will try to reproduce, but it would be easier for me if you could share some of your code, it seems that this is a different issue.

checkbox  unchecked 
![Clip_2024-10-09_12-45-28](https://github.com/user-attachments/assets/214e390b-f76e-45e0-91a1-f93df397374c)
![Clip_2024-10-09_12-52-33](https://github.com/user-attachments/assets/0e9df2f7-a210-416e-b928-9b8f2c13f51e)

**Reply by beto-rodriguez** [reply, marked as answer]
I'll take a look into this soon.

You are creating a new series instance everytime, it seems that it causes the issue, alternatively, you could just toogle the Series.IsVisible property, I think that should fix your issue, and also it should be more efficient.

**Reply by rocinn** [reply]
> I'll take a look into this soon.
> 
> You are creating a new series instance everytime, it seems that it causes the issue, alternatively, you could just toogle the Series.IsVisible property, I think that should fix your issue, and also it should be more efficient.

I'll give it a try.

**Reply by rocinn** [reply]
> I'll take a look into this soon.
> 
> You are creating a new series instance everytime, it seems that it causes the issue, alternatively, you could just toogle the Series.IsVisible property, I think that should fix your issue, and also it should be more efficient.

The `Series.IsVisible` property can solve this issue (but the legend at the bottom will display all the lines), it's just that colors with high distinguishability will continue to be occupied. The more lines there are, the smaller the color distinguishability will become.
![PixPin_2024-10-09_13-42-49](https://github.com/user-attachments/assets/f8b8f478-2436-48fd-b43f-b70871d62ec4)

**Reply by rocinn** [reply]
> I'll take a look into this soon.
> 
> You are creating a new series instance everytime, it seems that it causes the issue, alternatively, you could just toogle the Series.IsVisible property, I think that should fix your issue, and also it should be more efficient.

I did not use the colors provided by the control by default because they are not enough. When there is a shortage of colors with high distinguishability, I randomly generate colors that do not exist. The generated colors may have less distinguishability from the existing ones (which is, of course, my issue).

![Clip_2024-10-09_13-51-08](https://github.com/user-attachments/assets/8ddf28c7-bf56-46fd-b380-388840617663)

**Reply by beto-rodriguez** [reply]
@rocinn issue raised, Ill take a look and let you know on #1651
